src/ui/DynamicFacet/DynamicFacet.ts

```diff
@@ -587,6 +587,7 @@ export class DynamicFacet extends Component implements IAutoLayoutAdjustableInsi
     this.position = index + 1;
 
     this.onQueryResponse(response);
+    this.updateQueryStateModel();
```

Comment:
I'm curious how the auto-selection feature updating the QSM affects the history manager.

My guess is pressing "back" will set the state of the query without the facet selections. Once that query succeeds though, if we update the QSM on success, then it might reselect the facet values, adding a new history entry. So could the user be stuck in a loop?

Reply:
Good point, I just remembered that API recently implemented the `preventAutoSelect` parameter for a value, triggering that parameter we the user deselects with the value should prevent that from being an issue

Reply:
@samisayegh  Adding you back as a reviewer because it's kind of an important change